src/Agent.Worker/Release/Artifacts/TfGitArtifact.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Build;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts.Definition;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts
+{
+    public class TfGitArtifact : AgentService, IArtifactExtension
+    {
+        public Type ExtensionType => typeof(IArtifactExtension);
+        public AgentArtifactType ArtifactType => AgentArtifactType.TFGit;
+
+        public async Task DownloadAsync(IExecutionContext executionContext, ArtifactDefinition artifactDefinition, string downloadFolderPath)
+        {
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNull(artifactDefinition, nameof(artifactDefinition));
+            ArgUtil.NotNullOrEmpty(downloadFolderPath, nameof(downloadFolderPath));
+
+            var gitArtifactDetails = artifactDefinition.Details as GitArtifactDetails;
+            ArgUtil.NotNull(gitArtifactDetails, nameof(gitArtifactDetails));
+
+            ServiceEndpoint endpoint = executionContext.Endpoints.FirstOrDefault((e => string.Equals(e.Name, gitArtifactDetails.RepositoryId, StringComparison.OrdinalIgnoreCase)));
```

Comment:
We never set the endpoint type.
Yes this will not work. This is just a port of the existing agent code. I was not aware of multiple artifacts for same definition feature until last friday. We should take up this work incrementally.